src/Domain/Domain.hpp

```diff
@@ -52,10 +52,17 @@ class Domain {
    * determined. For more information on setting up domains, see the
    * [domain creation tutorial](\ref tutorial_domain_creation).
    *
+   * Can specify either one set of boundary conditions for each Block or no
+   * boundary conditions at all.
    */
-  explicit Domain(std::vector<std::unique_ptr<domain::CoordinateMapBase<
-                      Frame::Logical, Frame::Inertial, VolumeDim>>>
-                      maps) noexcept;
+  explicit Domain(
+      std::vector<std::unique_ptr<domain::CoordinateMapBase<
+          Frame::Logical, Frame::Inertial, VolumeDim>>>
+          maps,
+      std::vector<DirectionMap<
```

Comment:
While squashing, add include for `DirectionMap` and `domain::BoundaryConditions::BoundaryCondition`.